ui/src/ScholarReader.tsx

```diff
@@ -451,6 +442,33 @@ class ScholarReader extends React.PureComponent<ScholarReaderProps, State> {
     const elUserAnnotationTypeContainer = document.getElementById(
       "scholarReaderAnnotationTypeSelect"
     );
+    const highlightedSymbols: Map<String, BoundingBox> = new Map();
+    if (
+      this.state.symbols !== null &&
+      this.state.mathMls !== null &&
+      this.state.selectedEntityType === "symbol" &&
+      this.state.selectedEntityId !== null
+    ) {
+      // we need these highlights to be in sorted
+      // order by highlight y value so we must use a Map.
```

Comment:
Why must we use a map? I think I know but I'm not sure.

Reply:
Need the keys to maintain their order